docs/docs/configuration.rst

```diff
@@ -55,7 +55,105 @@ and your bespin.yml would look something like::
 
 Where ``<options>`` are the options for that stack.
 
-.. note:: The location of the stack and params json files are configured by the
- ``stack_json`` and ``params_json`` options. 
-    
+.. note:: The location of the stack template file is configured by the
+   ``stack_json`` or ``stack_yaml`` option. The location of the params file is
+   configured by the ``params_json`` or ``params_yaml`` option. Alternatively
+   parameters can be specified inline (inside bespin.yml) using ``params_yaml``.
+
 .. show_specs::
+
+
+Formatter
+---------
+
+Configuration values may reference other parts of the config using 'replacement
+fields' surrounded by curly braces ``{}``. Nested values can be referenced
+using dot notation, eg: ``{foo.bar.quax}``.
+If you need to include a brace character in the literal text, it can be escaped
+by doubling: ``{{`` and ``}}``.
+
+Available fields:
+
+environment
+  Current environment name as a string
+
+environments.<env_name>.*
+  Environment mappings.
+
+  Environment fields includes:
+
+  account_id
+    Environment AWS account id
+
+  region
+    Environment AWS region
+
+stacks.<stack_name>.*
+  Stack mappings.
+  See `Stack <#Stack>`_ spec for more detail.
+
+tags.*
+  Tags mapping
+
+vars.*
+  Vars mapping
+
+__stack_name__
+  Current stack name as a string.
```

Comment:
this and `__stack__` and `__environment__` are only available inside the options for a stack because it's injected into the available options when the stack is read in